### PR TITLE
Fix kafka spec failures on travis.

### PR DESCRIPTION
**Pull request by pankajroark**
Kafka spec is pretty flaky on travis. Trying to fix it. At this point the purpose of this PR is to try changes on travis, not ready for review yet.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,12 +4,12 @@ matrix:
...
     - scala: 2.11.11
       jdk: oraclejdk8
-      script: umask 0022 && ./sbt ++$TRAVIS_SCALA_VERSION clean coverage test coverageReport mimaReportBinaryIssues
+      script: umask 0022 && ./sbt ++$TRAVIS_SCALA_VERSION clean coverage storehaus-kafka/test coverageReport mimaReportBinaryIssues
       after_success:
         - bash <(curl -s https://codecov.io/bash)
     - scala: 2.12.2 
       jdk: oraclejdk8
-      script: umask 0022 && ./sbt ++$TRAVIS_SCALA_VERSION clean coverage test coverageReport mimaReportBinaryIssues
+      script: umask 0022 && ./sbt ++$TRAVIS_SCALA_VERSION clean coverage storehaus-kafka/test coverageReport mimaReportBinaryIssues
```
Comment (johnynek) [inline .travis.yml]: so we need to make sure not to merge with this setting.

Comment (pankajroark) [reply .travis.yml]: yes, this is just for trying out faster.

**Comment by pankajroark** [comment]
I think I've found the root cause and fixed it, please take a look. The issue was that on first poll after writing events to Kafka we may not get all. We need to try again a few times if we don't get all events.

**Comment by pankajroark** [comment]
Unfortunately, it looks like even this does not remove the flakiness completely, even though does seem to help.

**Comment by johnynek** [comment]
Thanks for working to get the tests less flaky.

**Comment by pankajroark** [comment]
I think the Kafka tests are much better now. Looking forward to merging this.

**Comment by johnynek** [comment]
:+1:

Next we should use codecov to see where we can easily improve test coverage.

**Review by johnynek — APPROVED** [review]

**Pull request merged**